spring-boot-project/spring-boot-docs/src/docs/antora/modules/reference/pages/messaging/pulsar.adoc

```diff
@@ -150,11 +150,11 @@ include-code::MyBean[]
 Spring Boot auto-configuration provides all the components necessary for `PulsarListener`, such as the `PulsarListenerContainerFactory` and the consumer factory it uses to construct the underlying Pulsar consumers.
 You can configure these components by specifying any of the `spring.pulsar.listener.\*` and `spring.pulsar.consumer.*` prefixed application properties.
 
-If you need more control over the consumer factory configuration, consider registering one or more `ConsumerBuilderCustomizer` beans.
+If you need more control over the configuration of the consumer factory used by the container factory to create consumers, consider registering one or more `ConsumerBuilderCustomizer` beans.
 These customizers are applied to all consumers created by the factory, and therefore all `@PulsarListener` instances.
 You can also customize a single listener by setting the `consumerCustomizer` attribute of the `@PulsarListener` annotation.
 
-
+If you need more control over the actual container factory configuration, consider registering one or more `PulsarContainerFactoryCustomizer<ConcurrentPulsarListenerContainerFactoryCustomizer<?>>` beans.
```

Comment:
I chose to describe how to customize only the 3 concrete listener container factories.
I left out the fact that they can actually customize anywhere up/down the `PulsarContainerFactory` hierarchy as the listener hierarchy is a bit noisy and I think it may confuse things.

Reply:
I don't have an opinion, except the mouthful name there is unfortunate (there's an extra customizer in the generic that I've removed).